From 49aa7370cc78565efd3ce25ef7d625bbc3f496bc Mon Sep 17 00:00:00 2001
From: vdanix <vishwanathx.dani@intel.com>
Date: Wed, 8 Dec 2021 11:05:27 +0530
Subject: [PATCH] Revert "INTERNAL: Revert "INTERNAL: Remove INVALID_OPERATION
 error generation""

This reverts commit aba94df2819594c9dab9c6f361393fd6e07db09a.
---
 src/mesa/main/framebuffer.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/mesa/main/framebuffer.c b/src/mesa/main/framebuffer.c
index a5460795b88..07cb925e4d1 100644
--- a/src/mesa/main/framebuffer.c
+++ b/src/mesa/main/framebuffer.c
@@ -824,9 +824,12 @@ _mesa_get_color_read_format(struct gl_context *ctx,
        * we will assume that on that situation the same error should be
        * triggered too.
        */
-      _mesa_error(ctx, GL_INVALID_OPERATION,
-                  "%s(GL_IMPLEMENTATION_COLOR_READ_FORMAT: no GL_READ_BUFFER)",
-                  caller);
+
+      /*
+	   * IMPLEMENTATION_COLOR_READ_FORMAT is for optimization purpose. if return
+	   * GL_NONE, just indicated there was no optimized additional format-type
+	   * can be used. So we decide to remove INVALID_OPERATION error generation.
+	   */
       return GL_NONE;
    }
    else {
@@ -899,9 +902,6 @@ _mesa_get_color_read_type(struct gl_context *ctx,
       /*
        * See comment on _mesa_get_color_read_format
        */
-      _mesa_error(ctx, GL_INVALID_OPERATION,
-                  "%s(GL_IMPLEMENTATION_COLOR_READ_TYPE: no GL_READ_BUFFER)",
-                  caller);
       return GL_NONE;
    }
    else {
-- 
2.34.1

